api/app_analytics/tasks.py

```diff
@@ -0,0 +1,151 @@
+        host=host,
+    )
+
+
+def _get_api_usage_source_data(
+    process_from: datetime, process_till: datetime, source_bucket_size: int = None
+) -> dict:
+    filters = Q(
+        created_at__lte=process_till,
+        created_at__gt=process_from,
+    )
+    if source_bucket_size:
+        return (
+            APIUsageBucket.objects.filter(filters, bucket_size=source_bucket_size)
+            .values("environment_id", "resource")
+            .annotate(count=Sum("total_count"))
+        )
+    return (
+        APIUsageRaw.objects.filter(filters)
+        .values("environment_id", "resource")
+        .annotate(count=Count("id"))
+    )
+
+
+def _get_feature_evaluation_source_data(
+    process_from: datetime, process_till: datetime, source_bucket_size: int = None
+) -> dict:
+    filters = Q(
+        created_at__lte=process_till,
+        created_at__gt=process_from,
+    )
+    if source_bucket_size:
+        return (
+            FeatureEvaluationBucket.objects.filter(
+                filters, bucket_size=source_bucket_size
+            )
+            .values("environment_id", "feature_name")
+            .annotate(count=Sum("total_count"))
+        )
+    return (
+        FeatureEvaluationRaw.objects.filter(filters, bucket_size=source_bucket_size)
+        .values("environment_id", "feature_name")
+        .annotate(count=Sum("evaluation_count"))
+    )
+
+
+def get_start_of_current_bucket(bucket_size: int) -> datetime:
+    current_time = timezone.now().replace(second=0, microsecond=0)
+    start_of_current_bucket = current_time - timezone.timedelta(
+        minutes=current_time.minute % bucket_size
+    )
+    return start_of_current_bucket
+
+
+def populate_api_usage_bucket(
+    bucket_size: int, run_every: int, source_bucket_size: int = None
+):
+    start_of_current_bucket = get_start_of_current_bucket(
+        bucket_size,
+    )
+
+    for i in range(1, (run_every // bucket_size) + 1):
+        process_from = start_of_current_bucket - timezone.timedelta(
+            minutes=bucket_size * i
+        )
+        process_to = process_from + timezone.timedelta(minutes=bucket_size)
+        data = _get_api_usage_source_data(process_from, process_to, source_bucket_size)
+        for row in data:
+            APIUsageBucket.objects.create(
+                environment_id=row["environment_id"],
+                resource=row["resource"],
+                total_count=row["count"],
+                bucket_size=bucket_size,
+                created_at=process_from,
+            )
+
+
+if settings.USE_POSTGRES_FOR_ANALYTICS:
+
+    @register_task_handler(run_every=60)
+    def populate_bucket_size_task(
+        bucket_size: int, run_every: int, source_bucket_size: int = None
+    ):
+        populate_api_usage_bucket(bucket_size, run_every, source_bucket_size)
+        populate_feature_evaluation_bucket(bucket_size, run_every, source_bucket_size)
+
+    populate_bucket_size_task.delay(settings.ANALYTICS_READ_BUCKET_SIZE, 60)
```

Comment:
But we still need to pass the args/kwargs that the function needs, another way would be to use a different decorator that does that

Reply:
Hmm... yes, interesting. I like the idea of using the decorator instead. 

I think actually, in reviewing this again, this call to `delay` is incorrect? I think it needs to `args` and/or `kwargs` as iterable / dict arguments rather than just passing the arguments directly like this.

Reply:
Yep, moved it to a decoartor